src/main/java/nz/ac/auckland/kafka/http/sink/request/ApiRequestInvoker.java

```diff
@@ -58,7 +63,27 @@ public void invoke(final Collection<SinkRecord> records){
             MDC.put(ApiRequest.REQUEST_HEADER_INFO_KEY, kafkaDetails);
 
             log.info("Processing record: topic={}  partition={} offset={} value={}", record.topic(), record.kafkaPartition(), record.kafkaOffset(), record.value().toString());
-            sendAPiRequest(record , traceId);
+            
+            Long sleepTime = 0L;
+            if (record.timestampType().equals(TimestampType.NO_TIMESTAMP_TYPE)){
+                log.info("Checking delay: now={}, sleep={}", start,new Date(start).toString(),new Date(this.config.requestDelay*1000).toString());
+                // If there is no timestamp we sleep for the specified delay
+                sleepTime = this.config.requestDelay * 1000L;
+            } else { 
+                long timePassed = start - record.timestamp();
+                log.info("Checking delay: now={}, record-timestamp={}, timepassed={}, sleep={}", new Date(start).toString(),new Date(record.timestamp()).toString(),timePassed,(this.config.requestDelay*1000 - timePassed));
+                if (timePassed < this.config.requestDelay) {
+                    sleepTime = (this.config.requestDelay * 1000L) - timePassed;
+                }
+            }
+
+            try {
+                Thread.sleep(sleepTime);
```

Comment:
Curious - my knowledge of Kafka is not incredibly strong - do you know if calling this without setting the `request.delay.sec` config i.e. calling `Thread.sleep(0);` would have any effect on event order/is that something that needs to be considered?

Reply:
To my understanding, [a single connector thread is launched when we create a Kafka connector](https://docs.confluent.io/platform/current/connect/devguide.html). Furthermore, when we create a sink connector, a consumer group is created for the given connector. 
The order the connector consumer group consumes records from a topic is based on an offset number. If the topic has only a single partition there is a global order of records within a topic. In this case, the consumer group (connector) reads records in the global order on a single thread and forwards them in the same order, sleeping or not. In the case of a topic having multiple partitions, we do not worry about the order as there is none.
I am also not a Kafka expert, so I could be wrong :)

Reply:
Sorry to be a pain but, could you move the try/catch contents into a `sendDelayedAPIRequest` for when the delay is not equal to zero, else just call the API. I'm keen to keep the behaviour as close as possible to how it's currently acting, and don't want to introduce any side effects due to my naivety around kafka/`Thread.sleep(0)` i.e.

```
if (delay == 0L) {
  sendApiRequest();
} else {
  sendDelayedApiRequest();
}
```

Happy to merge the PR after that since I can guarantee minimal behavioral changes.

Reply:
No worries, totally understand.